Core/ConnectedShapeBasedInterpolation.h

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) 2018 The Foundation for Research on Information Technologies in Society (IT'IS).
+ * 
+ * This file is part of iSEG
+ * (see https://github.com/ITISFoundation/osparc-iseg).
+ * 
+ * This software is released under the MIT License.
+ *  https://opensource.org/licenses/MIT
+ */
+
+#pragma once
+
+#include <itkCastImageFilter.h>
+#include <itkTranslationTransform.h>
+#include <itkNearestNeighborInterpolateImageFunction.h>
+#include <itkImageRegionConstIteratorWithIndex.h>
+#include <itkResampleImageFilter.h>
+#include <itkSignedMaurerDistanceMapImageFilter.h>
+#include <itkExtractImageFilter.h>
+#include <itkBinaryThresholdImageFilter.h>
+#include <itkConnectedComponentImageFilter.h>
+
+#include "itkSliceContiguousImage.h"
+
+#include <vector>
+
+namespace iseg {
+
+class ConnectedShapeBasedInterpolation
+{
+public:
+	using image_type = itk::Image<float, 2>;
+	using mask_type = itk::Image<unsigned char, 2>;
+	using image_stack_type = itk::SliceContiguousImage<float>;
+
+	std::vector<image_type::Pointer> interpolate(const image_type* slice1, const image_type* slice2, int number_of_slices)
+	{
+		int num_objects1 = 0;
+		int num_objects2 = 0;
+		auto objects1 = get_components(slice1, num_objects1);
+		auto objects2 = get_components(slice2, num_objects2);
+		if (num_objects1 != 1 || num_objects2 != 1)
+		{
+			throw std::runtime_error("Detected multiple foreground objects");
+		}
+
+		auto x1 = get_center_of_mass(objects1, num_objects1)[0];
```

Comment:
is the use of const in front of auto not part of the conventions? would make sense here...

Reply:
ok, will fix this